### PR TITLE
Add some light documentation

### PR DESCRIPTION
**Pull request by smithjessk**
Just a small PR to add some documentation for things that I found interesting when reading the code. :)

### CODE REVIEW

**Comment by smithjessk** [comment]
I noticed that there are lots of `HEAD` requests. These are used to [determine](https://github.com/oughtinc/ice/blob/bcd7963c0d13b89cdfa409fe681cc8e969eabb4a/ui/components/TracePage/TracePage.tsx#L52) the length of the [static trace files](https://github.com/oughtinc/ice/blob/bcd7963c0d13b89cdfa409fe681cc8e969eabb4a/ice/routes/traces.py#L32). 

It is possible and permitted, although I don't know if it's common, to append values to a trace after it already exists. That's why we can't cache `getContentLength`. Is all this true? Am I missing any important considerations?

**Comment by lslunis** [comment]
> I noticed that there are lots of `HEAD` requests. These are used to [determine](https://github.com/oughtinc/ice/blob/bcd7963c0d13b89cdfa409fe681cc8e969eabb4a/ui/components/TracePage/TracePage.tsx#L52) the length of the [static trace files](https://github.com/oughtinc/ice/blob/bcd7963c0d13b89cdfa409fe681cc8e969eabb4a/ice/routes/traces.py#L32).
> 
> It is possible and permitted, although I don't know if it's common, to append values to a trace after it already exists. That's why we can't cache `getContentLength`. Is all this true? Am I missing any important considerations?

Discussed off-thread. Summary: It's common to append to a trace after it already exists, but this whole mechanism of doing fake streaming ought to be replaced.

File: ice/utils.py [hunk 1]
```diff
@@ -75,6 +75,12 @@ def merge_strategy(
     return merge_strategy(merge_strategy, [], base, nxt)
 
 
+In = TypeVar("In", covariant=True)
+Out = TypeVar("Out", covariant=True)
```
Comment (lslunis) [inline ice/utils.py:79]: Are these unused?

**Review by lslunis — APPROVED** [review]

**Comment by stuhlmueller** [comment]
Closing for now, feel free to reopen.

**Pull request closed without merging**